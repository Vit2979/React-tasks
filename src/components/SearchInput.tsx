import React from 'react';

interface SearchInputProps {
  searchQuery: string;
  onSearchQueryChange: (query: string) => void;
  onSearchClick: () => void;
}

class SearchInput extends React.Component<SearchInputProps> {
  componentDidMount() {
    this.props.onSearchClick();
  }

  render() {
    const { searchQuery, onSearchQueryChange, onSearchClick } = this.props;

    return (
      <div id="top-section">
        <input
          type="text"
          value={searchQuery}
          onClick={() => onSearchQueryChange('')}
          onChange={(e) => onSearchQueryChange(e.target.value)}
        />
        <button type="button" onClick={onSearchClick}>
          Search
        </button>
      </div>
    );
  }
}

export default SearchInput;