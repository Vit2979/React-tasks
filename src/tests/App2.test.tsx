import React from 'react';
import { render } from '@testing-library/react';
import SearchInput from '../components/SearchInput';

describe('SearchInput', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(
      <SearchInput
        searchQuery=""
        onSearchQueryChange={() => {}}
        onSearchClick={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
