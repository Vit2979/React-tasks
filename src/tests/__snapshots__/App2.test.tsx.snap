// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="top-section"
  >
    <input
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
